http-body-util/src/buffered.rs

```diff
@@ -0,0 +1,112 @@
+use bytes::{Buf, Bytes};
+use http::HeaderMap;
+use http_body::Frame;
+
+use crate::util::BufList;
+
+/// A buffered body produced by [`BodyExt::buffered`] which collects all the DATA frames
+/// and trailers.
+#[derive(Debug)]
+pub struct Buffered<B> {
```

Comment:
@seanmonstar Should we `impl Body` on this type as well? I think it could just return `Data = B` and have an `Infallible` error?